metadata-ingestion/src/datahub/ingestion/api/auto_work_units/auto_dataset_properties_aspect.py

```diff
@@ -0,0 +1,102 @@
+import dataclasses
+from typing import Dict, Iterable, Optional
+
+from datahub.ingestion.api.workunit import MetadataWorkUnit
+from datahub.metadata.schema_classes import (
+    DatasetPropertiesClass,
+    OperationClass,
+    TimeStampClass,
+)
+from datahub.specific.dataset import DatasetPatchBuilder
+
+
+@dataclasses.dataclass
+class TimestampPair:
+    last_modified: Optional[TimeStampClass]
+    last_updated_timestamp: Optional[int]
+
+
+def auto_patch_last_modified(
+    stream: Iterable[MetadataWorkUnit],
+) -> Iterable[MetadataWorkUnit]:
+    """
+    Generate a patch request for datasetProperties aspect in-case
+        1. `lastModified` of datasetProperties is not set
+        2. And there are operation aspects
+    in this case set the `lastModified` of datasetProperties to max value of operation aspects `lastUpdatedTimestamp`.
+
+    We need this functionality to support sort by `last modified` on UI.
+    """
+    candidate_dataset_for_patch: Dict[str, TimestampPair] = {}
+
+    for wu in stream:
+        if not wu.get_urn().startswith("urn:li:dataset:"):
+            yield wu
+            continue
+
+        dataset_properties_aspect = wu.get_aspect_of_type(DatasetPropertiesClass)
+        dataset_operation_aspect = wu.get_aspect_of_type(OperationClass)
+
+        timestamp_pair = candidate_dataset_for_patch.get(wu.get_urn())
+
+        if timestamp_pair:
+            # Update the timestamp_pair
+            if dataset_properties_aspect and dataset_properties_aspect.lastModified:
+                timestamp_pair.last_modified = dataset_properties_aspect.lastModified
+
+            if (
+                dataset_operation_aspect
+                and dataset_operation_aspect.lastUpdatedTimestamp
+            ):
+                if timestamp_pair.last_updated_timestamp:
+                    # lastUpdatedTimestamp is present and less than the current operation aspect lastUpdateTimestamp
+                    # than update the timestamp_pair
+                    if (
+                        timestamp_pair.last_updated_timestamp
+                        < dataset_operation_aspect.lastUpdatedTimestamp
+                    ):
+                        timestamp_pair.last_updated_timestamp = (
```

Comment:
use `pair.last_updated_timestamp = max(pair. last_updated_timestamp or 0, operation.field)`

that will let you avoid much of this if...else complexity